bundler/lib/bundler/compact_index_client/cache.rb

```diff
@@ -54,67 +63,41 @@ def versions_etag_path
         directory.join("versions.etag")
       end
 
-      def checksums
-        checksums = {}
-
-        lines(versions_path).each do |line|
-          name, _, checksum = line.split(" ", 3)
-          checksums[name] = checksum
-        end
-
-        checksums
-      end
-
-      def dependencies(name)
-        lines(info_path(name)).map do |line|
-          parse_gem(line)
-        end
-      end
-
       def info_path(name)
         name = name.to_s
         if /[^a-z0-9_-]/.match?(name)
           name += "-#{SharedHelpers.digest(:MD5).hexdigest(name).downcase}"
-          info_roots.last.join(name)
+          @special_characters_info_root.join(name)
         else
-          info_roots.first.join(name)
+          @info_root.join(name)
         end
       end
 
       def info_etag_path(name)
         name = name.to_s
-        info_etag_root.join("#{name}-#{SharedHelpers.digest(:MD5).hexdigest(name).downcase}")
+        @info_etag_root.join("#{name}-#{SharedHelpers.digest(:MD5).hexdigest(name).downcase}")
       end
 
-      private
-
-      def mkdir(dir)
-        SharedHelpers.filesystem_access(dir) do
-          FileUtils.mkdir_p(dir)
+      def mkdir(name)
+        directory.join(name).tap do |dir|
+          SharedHelpers.filesystem_access(dir) do
+            FileUtils.mkdir_p(dir)
+          end
         end
       end
 
-      def lines(path)
-        return [] unless path.file?
-        lines = SharedHelpers.filesystem_access(path, :read, &:read).split("\n")
-        header = lines.index("---")
-        header ? lines[header + 1..-1] : lines
-      end
-
-      def parse_gem(line)
-        @dependency_parser ||= GemParser.new
-        @dependency_parser.parse(line)
+      def read(path)
+        return unless path.file?
+        SharedHelpers.filesystem_access(path, :read, &:read)
       end
 
-      def info_roots
-        [
-          directory.join("info"),
-          directory.join("info-special-characters"),
-        ]
+      def fetch(remote, path, etag_path, &blk)
+        @updater.update url(remote), path, etag_path
+        read(path)
       end
 
-      def info_etag_root
-        directory.join("info-etags")
+      def url(remote)
+        remote
```

Comment:
is this necessary?

Reply:
Preserved only because it looked so weird. I figured it was a plugin attach point. This was a pure refactor so I tried to change nothing.

Reply:
Removed.